### PR TITLE
Fixes for various bugs/issues logged as a part of the test day.

### LINKED ISSUE #17184: Export CVE data is inaccurate
**Expected behavior and actual behavior:**
Expected:The exported CVE data is consistent with the UI page
Actual:The exported CVE data is inconsistent with the UI page.Package column displayed incorrectly.Missing CVE data

**Steps to reproduce the problem:**
1. Create a project as test02
2. Push images to this project
    docker push domain/test02/goharbor/harbor-portal:v2.5.3
    docker push domain/test02/goharbor/notary-server-photon:v2.5.3
3. Scan
4. Export CVE And download
图片

5. Compare CVE data
UI page Projects<test02<goharbor/notary-server-photon

图片

图片

Package column is not consistent with UI page, there are curl and curl-libs in UI page, only curl-libs in CSV file.
UI page has 8 records in curl + curl-libs and 7 records in CSV file.

**Versions:**

- harbor version: v2.6.0-f3edb03b

### LINKED ISSUE #17187: Add current version and fixed version columns in export CVE data csv_file
Currently it is not possible to distinguish between cases where the CVE ID and package are the same but the version is different.
图片

### LINKED ISSUE #17188: Export the same CVE ID and pakcage issue
**Expected behavior and actual behavior:**
Expected:Export CVE data is the same as the UI page
Actual:Export CVE data is different from UI page.
Two records with the same CVE id and the same package, but with different versions, will only show one record in the csv file.

**Steps to reproduce the problem:**
1. Create a project as test02
2. Push image(goharbor/notary-server-photon:v2.5.3) to test02
2. Export CVE data
图片
3. Compare CVE data
图片
csv_file

But even if the data in the csv_file is complete, they cannot be distinguished in the csv_file, because they differ only in the version, and we should add the version columns.
Open this issue(https://github.com/goharbor/harbor/issues/17187) to discuss add version columns.

**Versions:**
- harbor version: v2.6.0-f3edb03b

### LINKED ISSUE #17193: Export CVE data Filters issue
**Steps to reproduce the problem:**
1. Create new test01 and test02 projects
2. Push image(redis:5.0) to test01
3. Push image(nginx:latest) to test02
4. Create a new export label
图片

5. Add export label to test01/redis:5.0
6. Add export label to test02/nginx:latest
7. Scan and export CVE data
图片

**Expected behavior and actual behavior:**
Expected: Exporting redis without exporting nginx CVE data
Actual: Both redis and nginx CVE data are exported

**Versions:**
Please specify the versions of following systems.

- harbor version:  v2.6.0-f3edb03b

### LINKED ISSUE #17199: CVE export does not generate log information under /var/log/jobs/ in jobservice container
**Expected behavior and actual behavior:**
1. docker push x.x.x.x/library/nginx:1.13.12
2. get into the jobservice container shell: `docker exec -it <jobservice_container_id> /bin/bash`
3. within jobservice container shell: `ls /var/log/jobs/` there is no file
4. scan the nginx image, we can see there is a file generated with content under `/var/log/jobs/` in jobservice container: `ls /var/log/jobs/`
5. perform export cve against the `library` project, and cve data are exported and downloaded with cve information in it.
6. However, `ls /var/log/jobs/`, the new log file generated under `/var/log/jobs/` in jobservice container is empty (**actual behavior**). 
    This new file under `/var/log/jobs/` in jobservice container should contain some log information about the export cve log activity (**expected behavior**).
Screen Shot 2022-07-19 at 14 56 20

**Steps to reproduce the problem:**
Please provide the steps to reproduce this problem.
please check the above description

**Versions:**
Please specify the versions of following systems.
* Harbor Version: Version v2.6.0-f3edb03b

### PR DESCRIPTION
**Pull request by prahaladdarkin**
Closes #17199
Closes #17193
Closes #17188
Closes #17184
Closes #17187
Closes:
* CVE Data Export API IDOR issue
* https://github.com/goharbor/harbor/issues/17199
* https://github.com/goharbor/harbor/issues/17193
* https://github.com/goharbor/harbor/issues/17188
* https://github.com/goharbor/harbor/issues/17184

Signed-off-by: prahaladdarkin <prahaladd@vmware.com>

Thank you for contributing to Harbor!

# Comprehensive Summary of your change

# Issue being fixed
Fixes #(issue)

Please indicate you've done the following:
- [x] Well Written Title and Summary of the PR
- [x] Label the PR as needed. "release-note/ignore-for-release, release-note/new-feature, release-note/update, release-note/enhancement, release-note/community, release-note/breaking-change, release-note/docs, release-note/infra, release-note/deprecation"
- [x] Accepted the DCO. Commits without the DCO will delay acceptance.
- [x] Made sure tests are passing and test coverage is added if needed.
- [ ] Considered the docs impact and opened a new docs issue or PR with docs changes if needed in [website repository](https://github.com/goharbor/website).

### CODE REVIEW

File: src/server/v2.0/handler/scanexport.go [hunk 1]
```diff
@@ -198,6 +213,9 @@ func (se *scanDataExportAPI) GetScanDataExportExecutionList(ctx context.Context,
 	}
 	execs := make([]*models.ScanDataExportExecution, 0)
 	for _, execution := range executions {
+		if execution.UserName != secContext.GetUsername() {
```
Comment (chlins) [inline src/server/v2.0/handler/scanexport.go:216]: If line 210 `ListExecutions` by username, I think that function should guarantee the result so here we don't need to check again?

Comment (prahaladdarkin) [reply src/server/v2.0/handler/scanexport.go:216]: Agreed. This check is redundant. Removed.

**Review by chlins — APPROVED** [review]
lgtm

File: src/server/v2.0/handler/scanexport.go [hunk 2]
```diff
@@ -107,6 +107,21 @@ func (se *scanDataExportAPI) GetScanDataExportExecution(ctx context.Context, par
...
+		return se.SendError(ctx, err)
+	}
+
+	// check if the execution being fetched is owned by the current user
+	secContext, err := se.GetSecurityContext(ctx)
+	if err != nil {
+		return se.SendError(ctx, err)
+	}
+	if secContext.GetUsername() != execution.UserName {
+		return middleware.ResponderFunc(func(writer http.ResponseWriter, producer runtime.Producer) {
```
Comment (wy65701436) [inline src/server/v2.0/handler/scanexport.go:120]: return errors.New(nil).WithCode(errors.StatusForbiddenCode)

**Review by wy65701436 — APPROVED** [review]
lgtm

File: src/server/v2.0/handler/scanexport.go (hunk 2 above)
Comment (prahaladdarkin) [reply src/server/v2.0/handler/scanexport.go]: Done

**Pull request merged**